c_common/models/compressor_bit_field_models/src/compressor_includes/remove_default_routes.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2019-2020 The University of Manchester
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __REMOVE_DEFAULT_ROUTES_H__
+#define __REMOVE_DEFAULT_ROUTES_H__
+
+#include <stdbool.h>
+#include "bit_set.h"
+#include "../common/routing_table.h"
+
+//! \brief removes default routes from the routing tables
+//! \param[in] remove_elements: bool flag to actually remove elements from table
+//! \param[in/out] pointer_to_new_size: new size to set without removing
+//!     elements from table
+//! \return: whether successful or not
+static bool remove_default_routes_minimise(
+        int *pointer_to_new_size, bool remove_elements) {
+    // Mark the entries to be removed from the table
+    bit_set_t remove;
+    bool success = bit_set_init(&remove, routing_table_get_n_entries());
+    if (!success) {
+        log_debug("failed to initialise the bit_set. shutting down");
+        return false;
+    }
+
+    // Work up the table from the bottom, marking entries to remove
+    for (int i = routing_table_get_n_entries() - 1; i < 0; i--) {
```

Comment:
This line looks suspicious to me. I strongly doubt that `routing_table_get_n_entries()` will be returning a negative number!

Reply:
thats interesting. given thats not in the code we worked on. thats mundy's code..... but your right. it looks like the bool operator is reversed for some reason from what we'd expect......

might just be a use of switching the bool operator and seeing if it breaks anything

Reply:
given the orignal code was even more odd

https://github.com/project-rig/rig_routing_tables/blob/master/include/remove_default_routes.h#L15

im just testing with the correct bool logic to see if it breaks anything on the local machine

Reply:
Alan’s attempt at converting Dogie Mindy code that counted on uint32_t overflow to work.

Luckily not in a critical Bit of code.

Reply:
fixed